src/test/java/catan/controllers/ctf/Scenario.java

```diff
@@ -199,51 +216,109 @@ public Scenario nextRandomDiceValues(List<Integer> nextRandomValues) {
         return this;
     }
 
+    public Scenario nextRandomDevelopmentCards(List<DevelopmentCard> nextCards) {
+        for (DevelopmentCard nextCard : nextCards) {
+            randomUtil.setNextDevelopmentCard(nextCard);
+        }
+
+        return this;
+    }
+
     public HexBuilder setHex(HexType hexType, Integer diceValue) {
         return new HexBuilder(this, hexType, diceValue);
     }
 
-    private void saveUsersResourcesValues(){
-        if(!trackResources){
+    private void saveUsersResourcesAndCardsValues() {
+        if (!trackResources && !trackCards) {
             return;
         }
 
         getGameDetails(1);
-        int p1Brick = gameUser(1).getValueOf("resources.brick");
-        int p1Wood = gameUser(1).getValueOf("resources.wood");
-        int p1Sheep = gameUser(1).getValueOf("resources.sheep");
-        int p1Wheat = gameUser(1).getValueOf("resources.wheat");
-        int p1Stone = gameUser(1).getValueOf("resources.stone");
+        if (trackResources) {
+            int p1Brick = gameUser(1).getValueOf("resources.brick");
```

Comment:
i dont like that you use 'p' in the name of variables. is it the shortening of 'player'? in other places we use word 'user' (e.g. 244-248, 258-262 lines). I think that all namings should be in one style: either 'user' or 'player'